DynamicTablesPkg/Library/Common/SsdtPcieSupportLib/SsdtPcieOscTemplate.asl

```diff
@@ -46,21 +46,16 @@ DefinitionBlock ("SsdtPciOsc.aml", "SSDT", 2, "ARMLTD", "PCI-OSC", 1) {
 
       // Save Capabilities DWord2 & 3
       Store (CDW2, SUPP)
-      Store (CDW3, CTRL)
+      And (CTRL, CDW3, CTRL)
 
       // Only allow native hot plug control if OS supports:
       // * ASPM
       // * Clock PM
       // * MSI/MSI-X
       If (LNotEqual (And (SUPP, 0x16), 0x16)) {
-        And (CTRL, 0x1E, CTRL) // Mask bit 0 (and undefined bits)
+        And (CTRL, 0x9E, CTRL) // Mask bit 0 (and undefined bits)

```

Comment:
Can you explain the switch to mask `9E` here? Is this not keeping native hot plug control if the OS doesn't support the required features? Perhaps I am reading my bits incorrectly :)

Reply:
Chatted offline, I was reading my bits incorrectly. This masks off native hot plug control if the OS doesn't support it. With the new capability of setting more bits, we will change this to leave the other defined bits and only mask off native hotplug support.

Reply:
Hi,
Just for information, this piece of code comes from the 'PCI Firmware Specification Revision 3.3', s4.5.3. 'ASL Example'.

It seems that in revision 3.0, only bits [0:4] were defined (cf Table 4-5 and 4-6: Interpretation of the _OSC Control Field), and the example only saves these bits (except bit 0). Now that there are more bits defined, the example in the spec (and the original code in this file) might be out-dated.

Reply:
Thanks Pierre, this is a good call out. In revisiting the spec, I agree, this is an example for a limited mobile system, in order to genericize this, we need to ensure we are looking at the expanded bit requirements (like DPC for example) and enforcing them.